assets/src/block-editor/components/with-media-library-notice.js

```diff
@@ -57,47 +56,82 @@ export default ( InitialMediaUpload, minImageDimensions ) => {
 		 */
 		initFeaturedImage() {
 			const FeaturedImageSelectMediaFrame = getSelectMediaFrame( FeaturedImageToolbarSelect );
+
+			const FeaturedImageLibrary = wp.media.controller.FeaturedImage.extend( {
```

Comment:
I've extended ` wp.media.controller.FeaturedImage` instead since it already has all the required functionality for updating the featured image ID when being selected, and pushing the selected image to the front of the image collection when being shown in the modal .